### ISSUE TITLE
Implement Event(s) class structure

### ISSUE THREAD

**Issue by jsbean** [jsbean/HansTools#38]
Use base `Event` class, and subclass on a piece-specific basis.

**Comment by jsbean** [comment]
Given the YAML structure:

```YAML
1: 
  envelope: 
    filter: 0.8
    offset: 0.7
  soundfiles: 
    fleur: 
      envelope_to_gain: 0.8
      envelope_to_pitch: 0.9
      gain: 0.5
      use_envelope: true
    terre: 
      envelope_to_gain: 0.8
      envelope_to_pitch: -0.9
      gain: 0.6
      use_envelope: true
2: 
  envelope: 
    filter: 0.2
    offset: 0.0
  soundfiles: 
    blume: 
      gain: 0.5
      use_envelope: false
    erde: 
      envelope_to_gain: 0.9
      envelope_to_pitch: 3.9
      gain: 0.6
      use_envelope: true
```

For example, the implementation for `Voices Unrooted` can be:


```Swift
open class Event {

    struct Sound {
        let name: String
        // ...
    }

    public let soundName: String
    public let soundDuration: Double
    public let gain: Double
    public let fadeTime: Double
    public let countdown: Double
    public let tempo: Double
    public let progressBarTime: Double

    let index: Int
    let envelope: Float
    let soundFiles: [Sound]
    let progressBarTime: Float
    let tempo: Float
    let coundown: Float
    let fadeTime: Float
    let gain: Float
}
```

**Comment by jsbean** [comment]
@hanstutschku

**Comment by jsbean** [comment]
Closed by #57.
